LazyAfternoons/io-app
# feat: account deletion flow

## Short description
This PR adds the profile deletion flow which can be invoked from any point of the application. In this case, it can be started from the profile page by interacting with the "profile deletion status" switch or from the wallet by pressing the "delete profile" button.

## List of changes proposed in this pull request
- locales/en/index.yml & locales/it/index.yml: Adds locale strings for the deletion flow;
- ts/components/wallet/WalletHomeHeader.tsx: Adds a button to invoke the profile deletion flow;
- ts/navigation/ProfileNavigator.tsx: Adds the profile deletion flow screens;
- ts/navigation/params/ProfileParamsList.ts: Adds the profile deletion flow screens parameters;
- ts/navigation/routes.ts: Adds the profile deletion flow screens routes;
- ts/screens/profile/NewProfileScreen.tsx: Adds the `onChange` prop to the switch to invoke the profile deletion flow and refactors the usage of `getValueFromProfilePot`;
- ts/screens/profile/ProfileDeletionDetailsScreen.tsx: Adds a screen which shows details of the data which is going to be deleted along with the profile and again two buttons, the first one to cancel the operation and the second one to continue;
This screen also features a loading view, an error view with a `retry` button and a continue view which confirms that the request is being processed and closes the flow.;
- ts/screens/profile/ProfileDeletionScreen.tsx: Adds a screen which shows a warning for the user and two buttons, the first one to cancel the operation and the second one to continue;
- ts/utils/pot.ts: Moves the `getValueFromProfilePot` function here.

## How to test
- Point to the dev server;
- Generate locales via yarn generate, then start the app and navigate to Profile through the bottom navigator;
- Toggle the switch status to start the flow;
- The first screen should show an `InfoBox` and two buttons as explained above. The cancel button should make you return to the profile screen; 
- The second screen should show the details of the deletion and two buttons, as explained above. The cancel button should make you return to the profile screen;
- If everything goes as it should a new screen with a continue button should appear. The `continue` button should make you return to the profile screen; 
- The same applies when starting the flow from the wallet screen by pressing on the `Delete account` button.
